### PR TITLE
Genesis epoch

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Added an option to set the genesis epoch
- Added an option in chain simulator to provide an initial nonce
  
## Proposed changes
- Extended the `config.toml` file with the GenesisEpoch parameter
- Extended the chain simulator with and option to provide an initial nonce

## Testing procedure
- Ensure the internal testnet is working properly.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/node/config/config.toml [hunk 1]
```diff
@@ -621,6 +621,7 @@
    Type = "json"
 
 [EpochStartConfig]
+    GenesisEpoch = 0
```
Comment (iulianpascalau) [inline cmd/node/config/config.toml:624]: :+1:

File: node/chainSimulator/configs/configs.go [hunk 2]
```diff
@@ -117,6 +118,8 @@ func CreateChainSimulatorConfigs(args ArgsChainSimulatorConfigs) (*ArgsConfigsSi
 	configs.GeneralConfig.DbLookupExtensions.Enabled = true
 
 	configs.GeneralConfig.EpochStartConfig.ExtraDelayForRequestBlockInfoInMilliseconds = 1
+	configs.GeneralConfig.EpochStartConfig.GenesisEpoch = args.InitialEpoch
+	configs.EpochConfig.EnableEpochs.StakingV2EnableEpoch = args.InitialEpoch + 1
```
Comment (iulianpascalau) [inline node/chainSimulator/configs/configs.go:122]: We might add a TODO to fix the staking v2

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: genesis/process/genesisBlockCreator.go [hunk 3]
```diff
@@ -38,6 +38,9 @@ import (
 
 const accountStartNonce = uint64(0)
 
+var genesisNonce uint64
```
[the remark this answers is not in the document]
Comment (miiu96) [reply genesis/process/genesisBlockCreator.go]: fixed

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: v1.6.15-dev-config-eb2e06c06d -> configurable-epoch-a31eafa7fd

--- Specific errors ---

block hash does not match 157
wrong nonce in block 144
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  0
Nr. of all WARNS:  233
Nr. of new ERRORS:  0
Nr. of new WARNS:  1
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

--- WARNINGS ---

up-mad-validator-27 :
Warn: trie recreate error:           error = getNodeFromDB error: key not found for key root =  1

/------/

**Pull request merged**